xcm/xcm-executor/src/assets.rs

```diff
@@ -190,6 +251,92 @@ impl Assets {
 			match asset {
 				MultiAsset::None => (),
 				MultiAsset::All => return self.swapped(Assets::default()),
+				MultiAsset::AllFungible => {
+					// Copy all fungible assets into result.
+					for (id, amount) in self.fungible.clone().into_iter() {
+						result.saturating_subsume_fungible(id, amount);
+					}
+					// Clear all fungible assets.
+					self.fungible = Default::default();
+				},
+				MultiAsset::AllNonFungible => {
+					// Copy all non-fungible assets into result.
+					for (class, instance) in self.non_fungible.clone().into_iter() {
+						result.saturating_subsume_non_fungible(class, instance);
+					}
+					// Clear all non-fungible assets.
+					self.non_fungible = Default::default();
+				},
+				MultiAsset::AllAbstractFungible { id } => {
+					let all_abstract_fungible: Vec<(AssetId, u128)> = self.fungible.clone()
+						.into_iter()
+						.filter(|(iden, _)| match iden {
+							AssetId::Abstract(iden) => id == *iden,
+							_ => false,
+						}).collect();
```

Comment:
Here we have the same as above, but with additional `.collect` which would allocate a new vector. I believe that is not necessary since you destruct that vector by iterating on it just a line below (such a `for` loop calls `into_iter` under the hood).